tasks/gulp/clean.js

```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+const gulp = require('gulp')
+const taskArguments = require('./task-arguments')
+const del = require('del')
+
+// Clean task for a specified folder --------------------
+// Removes all old files, except for package.json
+// and README in all package
+// '/**/**/*' is to remove folder deep down  in globals/
+// ------------------------------------------------------
+const isPackages = (taskArguments.destination === 'packages') || false
+
+gulp.task('clean', () => {
+  return del.sync(isPackages ? [taskArguments.destination + '/**/*.*', taskArguments.destination + '/**/**/*', '!' + taskArguments.destination, '!' + taskArguments.destination + '/**/package.json', '!' + taskArguments.destination + 'all/README.md'] : [taskArguments.destination + '/**/*'])
```

Comment:
This is really hard to follow. Ternary doesn't feel like the right choice when you end up with the `:` hidden in the middle of what's already a big list.

I'd suggest perhaps something like:

```
gulp.task('clean', () => {
  const destination = taskArguments.destination

  if (destination === 'packages') {
    # Clean packages folder
    return del.sync([
      `${destination}/**/*.*`,
      `${destination}/**/**/*`,
      `!${destination}`,
      `!${destination}/**/package.json`,
      `!${destination}/all/README.md`
    ])
  } else {
    # Clean dist
    return del.sync([`${destination}/**/*`])
  }
})
```

Alternatively, it might be better just to define separate clean tasks for `clean:packages` and `clean:dist`? That would allow for more specific documentation as well.

Reply:
if inside the gulp task isn't really a thing unfortunately

Reply:
We're inside a function and using `return` here so are you sure that's the case?

Reply:
yeah tried both your example above and just using boolean.
I can try and move this up to a function outside and run the function on return

Reply:
I've got an example working locally which uses an `if` within the function and it works just fine. Happy to share the code if helpful.